src/main.rs

```diff
@@ -720,6 +730,62 @@ fn heltec_hello_world(
     })
 }
 
+#[cfg(feature = "ssd1306g_spi")]
+fn ssd1306g_hello_world_spi(
+    dc: gpio::Gpio4<gpio::Unknown>,
+    rst: gpio::Gpio16<gpio::Unknown>,
+    spi: spi::SPI3,
+    sclk: gpio::Gpio18<gpio::Unknown>,
+    sdo: gpio::Gpio23<gpio::Unknown>,
+    cs: gpio::Gpio5<gpio::Unknown>,
+) -> Result<()> {
+    info!("About to initialize the SSD1306 SPI LED driver");
+
+    let config = <spi::config::Config as Default>::default()
+        .baudrate(10.MHz().into())
+        .bit_order(spi::config::BitOrder::MSBFirst);
+
+    let di = SPIInterfaceNoCS::new(
+        spi::Master::<spi::SPI3, _, _, _, _>::new(
+            spi,
+            spi::Pins {
+                sclk,
+                sdo,
+                sdi: Option::<gpio::Gpio19<gpio::Unknown>>::None,
+                cs: Some(cs),
+            },
+            config,
+        )?,
+        dc.into_output()?,
+    );
+
+    let mut delay = delay::Ets;
+    let mut reset = rst.into_output()?;
+
+    reset.set_high()?;
```

Comment:
Since the `sdd1306` crate contains already a reset functionality, as you can see in the examples of the repo, i.e. https://github.com/jamwaffles/ssd1306/blob/master/examples/graphics.rs#L80, you could have replaced this manual reset logic with the more clear reset call.